### PR TITLE
Create auto-comment.yml

### PR DESCRIPTION
**Pull request by Nishitbaria**
**I have created a Workflow that automatically commits when an issue is opened, a pull request is created, an issue is closed, or an issue is assigned or unassigned.** 

 **` The issue number that resolves this particular issue is`:https://github.com/PiyushKalyanpy/GyanaGuru/issues/259**

too run this workflow you need to enable the following permission 

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Auto Comment on Issues Opened
+        uses: wow-actions/auto-comment@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          issuesOpened: |
+            Hey @{{ author }},
+
+            Thank you for raising an issue. We will investigate into the matter and get back to you as soon as possible.
```
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:24]:
```suggestion
            Thank you for raising this issue. We will investigate into the matter and get back to you as soon as possible.
```

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+            Please make sure you have given us as much context as possible.
+
+      - name: Auto Comment on Issues Closed
+        uses: wow-actions/auto-comment@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          issuesClosed: |
+            Hey @{{ author }}, This issue is closed.
```
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:33]: Can we also show the reason of issue closing?

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Auto Comment on Pull Request Merged
+        uses: wow-actions/auto-comment@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          pullRequestMerged: |
+            Hey, @{{ author }} Thank you for the valuable contribution!
```
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:40]:
```suggestion
            Hey @{{ author }}, thank you for your valuable contribution!
```

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml (hunk 2 above)
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:33]:
```suggestion
            Hey @{{ author }}, this issue is closed.
```

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Auto Comment on Pull Request Opened
+        uses: wow-actions/auto-comment@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          pullRequestOpened: |
+            Hey @{{ author }},
+            <br>
```
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:48]: Do we need this `<br>` here, if yes then we also need this in `issuesOpened`?

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            Thank you for raising your pull request and contributing to our Community 🎉
+
+            Please make sure you have followed our contributing guidelines. We will review it as soon as possible.
+
+      - name: Auto Comment on Issues Assigned
+        uses: wow-actions/auto-comment@v1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          issuesAssigned: |
+            Hey @{{ author }}, thank you for raising an issue. I have assigned the issue to you. You can now start working on it. If you encounter any problems, feel free to connect with us.
```
Comment (lalitmee) [inline .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:58]:
```suggestion
            Hey @{{ author }}, thank you for raising this issue. 
            I have assigned the issue to you.
            You can now start working on it. If you encounter any problems, feel free to connect with us.
```

**Review by lalitmee — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/workflows/auto-comment.yml (hunk 2 above)
Comment (Nishitbaria) [reply .github/ISSUE_TEMPLATE/workflows/auto-comment.yml:33]: yes we can Like If sone Who is not working on the issue then the issue is automatically closed then we can say that you are not working on that

**Review by lalitmee — COMMENTED** [review]
How do we know that it's working as expected? Have you tested it somehow?

**Comment by Nishitbaria** [comment]
@lalitmee Yes I have work in my Repository That works totally fine 😀

**Comment by Nishitbaria** [comment]
@lalitmee 

[screen-capture.webm](https://github.com/PiyushKalyanpy/GyanaGuru/assets/85815172/2e36e319-6d7e-4276-afc0-f241a99dc849)

**Pull request closed without merging**